src/main/java/mate/controller/cars/DeleteCarController.java

```diff
@@ -0,0 +1,23 @@
+package mate.controller.cars;
+
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.service.CarService;
+
+@WebServlet(urlPatterns = "/cars/delete")
+public class DeleteCarController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("mate");
+    private CarService carService = (CarService) injector.getInstance(CarService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        carService.delete(Long.valueOf(req.getParameter("id")));
```

Comment:
Please implement delete action to be performed in POST method. GET is for obtaining data, not modification

Reply:
When I implement delete action to be performed in POST method such error 405 occured:  HTTP method GET is not supported by this URL. The method received in the request-line is known by the origin server but not supported by the target resource. Also it is said, that I should use doGet for delete actions and doPost for update actions in this tutorial video: https://mate.academy/learn/java-web/java-web-practice?section=video&testTaskSlug=java-web-practice-hw&videoId=1007

Reply:
GET method is used in tutorial video as it's much easier to be presented.
Along with method change in controller, you should also change corresponding jsp page so your button will call POST method to backend and will pass id not in URL but in input parameter